deploy/tests/test_hub_health.py

```diff
@@ -0,0 +1,90 @@
+from pathlib import Path
+import os
+import pytest
+import backoff
+from jhub_client.execute import execute_notebook, JupyterHubAPI
+
+
+@pytest.fixture
+def notebook_dir(hub_type):
+    return Path(__file__).parent / 'test-notebooks' / hub_type
+
+def failure_handler(details):
+    print(f"Hub check health validation failed, hub not healthy.")
+
+def success_handler(details):
+    print(f"Notebook execution finished successfully.")
+
+
+
+# Try 2 times before declaring it a failure
```

Comment:
We're only trying on time now. Sorry for failing to update/remove the comment when introducing this change :(